storage/innobase/include/clone0api.h

```diff
@@ -0,0 +1,256 @@
+#endif
+
+/** Change cloned file states during recovery.
+@param[in]      finished        if recovery is finishing */
+void clone_files_recovery(bool finished);
+
+/** Initialize Clone system
+@return inndodb error code */
+dberr_t clone_init();
+
+/** Uninitialize Clone system */
+void clone_free();
+
+/** Check if active clone is running.
+@return true, if any active clone is found. */
+bool clone_check_active();
+
+/** @return true, if clone provisioning in progress. */
+bool clone_check_provisioning();
+#endif /* !UNIV_HOTBACKUP */
+
+/** Clone Notification handler. */
+class Clone_notify {
+ public:
+  /** Notification type. Currently used by various DDL commands. */
+  enum class Type {
+    /* Space is being created. */
+    SPACE_CREATE,
+    /* Space is being dropped. */
+    SPACE_DROP,
+    /* Space is being renamed. */
+    SPACE_RENAME,
+    /* Space is being discarded or imported. */
+    SPACE_IMPORT,
+    /* Space encryption property is altered. */
+    SPACE_ALTER_ENCRYPT,
+    /* Space encryption property of general tablespace is altered. */
+    SPACE_ALTER_ENCRYPT_GENERAL,
+    /* Space encryption flags of general tablespace are altered. */
+    SPACE_ALTER_ENCRYPT_GENERAL_FLAGS,
+    /* In place Alter general notification. */
+    SPACE_ALTER_INPLACE,
+    /* Inplace Alter bulk operation. */
+    SPACE_ALTER_INPLACE_BULK,
+    /* Special consideration is needed for UNDO as these DDLs
+    don't use DDL log and needs special consideration during recovery. */
+    SPACE_UNDO_TRUNCATE,
+    /* Redo log resizing */
+    SYSTEM_REDO_RESIZE
+  };
+
+#ifdef UNIV_HOTBACKUP
+  Clone_notify(Type, space_id_t, bool) : m_error() {}
+  ~Clone_notify() {}
+#else
+  /** Constructor to initiate notification.
+  @param[in]    type    notification type
+  @param[in]    space   tablespace ID for which notification is sent
+  @param[in]    no_wait set error and return immediately if needs to wait */
+  Clone_notify(Type type, space_id_t space, bool no_wait);
+
+  /** Destructor to automatically end notification. */
+  ~Clone_notify();
+#endif /* UNIV_HOTBACKUP */
+
+  /** Get notification message for printing.
+  @param[in]    begin    true if notification begin otherwise end
+  @param[out]   mesg    notification message */
+  void get_mesg(bool begin, std::string &mesg);
+
+  /** @return true iff notification failed. */
+  bool failed() const { return m_error != 0; }
+
+  /** @return saved error code. */
+  int get_error() const { return m_error; }
+
+  /** Disable copy construction */
+  Clone_notify(Clone_notify &) = delete;
+
+  /** Disable assignment */
+  Clone_notify &operator=(Clone_notify const &) = delete;
+
+ private:
+  /** Notification wait type set. */
+  enum class Wait_at {
+    /* Clone doesn't need to wait. */
+    NONE,
+    /* Clone needs to wait before entering. */
+    ENTER,
+    /* Clone needs to wait before state change. */
+    STATE_CHANGE,
+    /* Clone needs to abort. */
+    ABORT
+  };
```

Comment:
Each value is missing Doxygen comments. Are any other values than `ABORT` actually assigned for now? If not, the `Wait_at` as well as `m_wait` need to be `#if 0`’d or removed.